lib/pbench/server/database/models/users.py

```diff
@@ -119,6 +120,9 @@ def delete(username):
         """
         try:
             user_query = Database.db_session.query(User).filter_by(username=username)
+            if user_query.count() == 0:
+                raise NoResultFound(f"User {username} does not exist")
+
             user_query.delete()
             Database.db_session.commit()
             return True
```

Comment:
This function now either returns `True` or raises an exception.  On that basis, there's no point in returning `True`.  (And, the `:return:` clause in the docstring should be updated, either way.)